src/components/views/messages/TextualEvent.js

```diff
@@ -27,8 +28,10 @@ export default class TextualEvent extends React.Component {
         mxEvent: PropTypes.object.isRequired,
     };
 
+    static contextType = RoomContext;
+
     render() {
-        const text = TextForEvent.textForEvent(this.props.mxEvent);
+        const text = TextForEvent.textForEvent(this.props.mxEvent, this.context?.showHiddenEventsInTimeline);
```

Comment:
`?.` is a TS thing, we don't have the babel support for it to my knowedlege

Reply:
From what I can tell Babel has supported optional chaining [by default since 2019](https://github.com/babel/babel/pull/10811), and I see it's already used in quite a few places in JS code, so I don't think this should be a problem…

Reply:
@turt2live can you expand on this, you're my source on why we shouldn't be using it

Reply:
Hm, well then I think I'll take the route of merging `t3chguy/ts/8` (#6243) into this branch and converting the remaining couple of files in this PR to TypeScript, if that sounds okay?